src/__e2e__/bump.test.ts

```diff
@@ -1024,4 +1024,200 @@ describe('version bumping', () => {
     expect(modified).toContain('@beachball-comments-repro/package1');
     expect(modified).toContain('@beachball-comments-repro/package2');
   });
+
+  it('calls sync prebump hook before packages are bumped', async () => {
```

Comment:
Came back to this, to try to figure out the right thing to do. As a reference, prepublish and postpublish hooks throw, and are handled by the top-level exception handler in `cli.ts`.

The hook throwing terminating beachball seems like reasonable behavior. I don't think there is much reasonable we can do for rolling back the hooks, but it seems like it would be better to rollback the temporary publish branch on exception. Seems like a more generic problem though, to be fixed at a different layer...

For now I added tests that just verifies that exceptions are propagated.